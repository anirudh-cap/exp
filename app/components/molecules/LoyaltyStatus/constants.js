export const LIVE = 'live';
export const ENDED = 'ended';
export const INPROGRESS = 'inprogress';
export const INVALID = 'invalid';
export const TAG = 'tag';
export const YELLOW = 'yellow';
