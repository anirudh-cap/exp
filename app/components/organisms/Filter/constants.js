const FILTER_BY_OPTIONS = [
  { label: 'No filter', value: 'NO_FILTER' },
  { label: 'By Genre', value: 'BY_GENRE' },
  { label: 'By Author', value: 'BY_AUTHOR' },
  { label: 'By Name', value: 'BY_NAME' },
];

export { FILTER_BY_OPTIONS };
