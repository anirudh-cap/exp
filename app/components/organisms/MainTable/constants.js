export const HYPHEN = '-';

export const GET_PROGRAMS_REQUEST = 'loyaltyDetail/GET_PROGRAMS_REQUEST';
export const GET_PROGRAMS_SUCCESS = 'loyaltyDetail/GET_PROGRAMS_SUCCESS';
export const GET_PROGRAMS_FAILURE = 'loyaltyDetail/GET_PROGRAMS_FAILURE';

export const GET_USER_LIST_REQUEST = 'loyaltyDetail/GET_USER_LIST_REQUEST';
export const GET_USER_LIST_SUCCESS = 'loyaltyDetail/GET_USER_LIST_SUCCESS';
export const GET_USER_LIST_FAILURE = 'loyaltyDetail/GET_USER_LIST_FAILURE';

export const COLUMN_NAME = 'COLUMN_NAME';
export const COLUMN_TIRES = 'COLUMN_TIRES';
export const COLUMN_USER_WITH_TIME = 'COLUMN_USER_WITH_TIME';
export const COLUMN_KPIS = 'COLUMN_KPIS';
export const COLUMN_CONCEPTS_MAPPED = 'COLUMN_CONCEPTS_MAPPED';

export const GET_EMF_STATUS_REQUEST = 'loyaltyDetail/GET_EMF_STATUS_REQUEST';
export const GET_EMF_STATUS_SUCCESS = 'loyaltyDetail/GET_EMF_STATUS_SUCCESS';
export const GET_EMF_STATUS_FAILURE = 'loyaltyDetail/GET_EMF_STATUS_FAILURE';

export const CLEAR_DATA = 'loyaltyDetail/CLEAR_DATA';
export const EMF_ACTIVE = '1';

export const SORT_STR = 'SORT_STR';
export const SORT_DATE = 'SORT_DATE';
export const SORT_STATUS = 'SORT_STATUS';
