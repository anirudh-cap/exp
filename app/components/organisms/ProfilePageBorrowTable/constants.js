export const FETCH_USER_BORROWED_BOOKS = 'FETCH_USER_BORROWED_BOOKS';
export const FETCH_USER_BORROWED_BOOKS_SUCCESS =
  'FETCH_USER_BORROWED_BOOKS_SUCCESS';
export const FETCH_USER_BORROWED_BOOKS_FAILURE =
  'FETCH_USER_BORROWED_BOOKS_FAILURE';

export const RETURN_USER_BORROWED_BOOKS = 'RETURN_USER_BORROWED_BOOKS';
export const RETURN_USER_BORROWED_BOOKS_SUCCESS =
  'RETURN_USER_BORROWED_BOOKS_SUCCESS';
export const RETURN_USER_BORROWED_BOOKS_FAILURE =
  'RETURN_USER_BORROWED_BOOKS_FAILURE';
