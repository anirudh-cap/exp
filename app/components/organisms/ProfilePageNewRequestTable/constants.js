export const FETCH_USER_NEW_REQUESTED_BOOKS = 'FETCH_USER_NEW_REQUESTED_BOOKS';
export const FETCH_USER_NEW_REQUESTED_BOOKS_SUCCESS ='FETCH_USER_NEW_REQUESTED_BOOKS_SUCCESS';
export const FETCH_USER_NEW_REQUESTED_BOOKS_FAILURE ='FETCH_USER_NEW_REQUESTED_BOOKS_FAILURE';

export const CANCEL_USER_NEW_REQUESTED_BOOKS ='CANCEL_USER_NEW_REQUESTED_BOOKS';
export const CANCEL_USER_NEW_REQUESTED_BOOKS_SUCCESS ='CANCEL_USER_NEW_REQUESTED_BOOKS_SUCCESS';
export const CANCEL_USER_NEW_REQUESTED_BOOKS_FAILURE ='CANCEL_USER_NEW_REQUESTED_BOOKS_FAILURE';

export const ACCEPT_USER_NEW_BOOK_REQUEST ='ACCEPT_USER_NEW_BOOK_REQUEST';
export const ACCEPT_USER_NEW_BOOK_REQUEST_SUCCESS ='ACCEPT_USER_NEW_BOOK_REQUEST_SUCCESS';
export const ACCEPT_USER_NEW_BOOK_REQUEST_FAILURE ='ACCEPT_USER_NEW_BOOK_REQUEST_FAILURE';
