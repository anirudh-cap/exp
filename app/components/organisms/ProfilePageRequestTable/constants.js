export const FETCH_USER_REQUESTED_BOOKS = 'FETCH_USER_REQUESTED_BOOKS';
export const FETCH_USER_REQUESTED_BOOKS_SUCCESS =
  'FETCH_USER_REQUESTED_BOOKS_SUCCESS';
export const FETCH_USER_REQUESTED_BOOKS_FAILURE =
  'FETCH_USER_REQUESTED_BOOKS_FAILURE';
export const CANCEL_USER_REQUESTED_BOOKS = 'CANCEL_USER_REQUESTED_BOOKS';
export const CANCEL_USER_REQUESTED_BOOKS_SUCCESS =
  'CANCEL_USER_REQUESTED_BOOKS_SUCCESS';
export const CANCEL_USER_REQUESTED_BOOKS_FAILURE =
  'CANCEL_USER_REQUESTED_BOOKS_FAILURE';
