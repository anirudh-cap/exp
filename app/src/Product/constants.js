export const GET_PRODUCT_REQUEST = 'GET_PRODUCT_REQUEST';
export const GET_PRODUCT_SUCCESS = 'GET_PRODUCT_SUCCESS';
export const GET_PRODUCT_FAILURE = 'GET_PRODUCT_FAILURE';
export const GET_CATEGORY_REQUEST = 'GET_CATEGORY_REQUEST';
export const GET_CATEGORY_SUCCESS = 'GET_CATEGORY_SUCCESS';
export const GET_CATEGORY_FAILURE = 'GET_CATEGORY_FAILURE';
