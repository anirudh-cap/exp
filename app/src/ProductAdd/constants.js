export const SET_PRODUCT_REQUEST = 'SET_PRODUCT_REQUEST';
export const SET_PRODUCT_SUCCESS = 'SET_PRODUCT_SUCCESS';
export const SET_PRODUCT_FAILURE = 'SET_PRODUCT_FAILURE';
export const RESET_STATUS = 'RESET_STATUS';

export const blankForm = {
    title: '',
    description: '',
    brand: '',
    category: '',
    price: '',
    stock: '',
  };