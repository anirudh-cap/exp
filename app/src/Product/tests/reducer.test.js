import productReducer from '../reducer';
import * as types from '../constants';
import { fromJS } from 'immutable';

const error = new Error('error');

describe('getProductsReducer', () => {
  it('it handles the reducer with default type', () => {
    expect(productReducer(undefined, {})).toEqual(fromJS({
      error: null,
      products: null,
    }));
  });

  it('it handles the GET_PRODUCT_REQUEST action', () => {
    const action = {
      type: types.GET_PRODUCT_REQUEST,
    };
    expect(productReducer(undefined, action)).toEqual(fromJS({
        error: null,
        products: null,
    }));
  });

  it('it handles the GET_PRODUCT_SUCCESS action', () => {
    const action = {
      type: types.GET_PRODUCT_SUCCESS,
      payload: [],
    };
    expect(productReducer(undefined, action)).toEqual(fromJS({
      products:[],
      error:null,
    }));
  });

  it('it handles the GET_PRODUCT_FAILURE action', () => {
    const action = {
      type: types.GET_PRODUCT_FAILURE,
      payload:"failure",
    };
    expect(productReducer(undefined, action)).toEqual(fromJS({
      error:"failure",
      products:null,
    }));
  });
});
