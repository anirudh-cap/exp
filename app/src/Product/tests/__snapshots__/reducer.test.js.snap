// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getProductsReducer it handles the reducer with default type 1`] = `
Immutable.Map {
  "error": null,
  "products": null,
}
`;
